src/Catalog/Registration/RegistrationCollector.cs

```diff
@@ -24,7 +25,8 @@ public class RegistrationCollector : SortingGraphCollector
 
         private readonly StorageFactory _legacyStorageFactory;
         private readonly StorageFactory _semVer2StorageFactory;
-        private readonly ShouldIncludeRegistrationPackage _shouldIncludeSemVer2;
+        private readonly ShouldIncludeRegistrationPackage _shouldIncludeSemVer2ForLegacyStorageFactory;
+        private readonly RegistrationMakerCatalogItem.PostProcessGraph _postProcessGraphForLegacyStorageFactory;
```

Comment:
Why nest the delegate inside a class?

Reply:
I wanted to make it more obvious that it's strictly used by `RegistrationMakerCatalogItem`.
For example, where `ShouldIncludeRegistrationPackage` is used is much harder to find.